azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/VirtualMachineScaleSet.java

```diff
@@ -0,0 +1,110 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jclouds.azurecompute.arm.domain;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableMap;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+import java.util.Map;
+
+/**
+ * VirtualMachineScaleSet for subscription
+ */
+@AutoValue
+public abstract class VirtualMachineScaleSet {
+
+   /**
+    * The id of the virtual machine scale set
+    */
+   @Nullable
+   public abstract String id();
+
+   /**
+    * The name of the virtual machine scale set
+    */
+   @Nullable
+   public abstract String name();
+
+   /**
+    * The location of the virtual machine scale set
+    */
+   @Nullable
+   public abstract String location();
+
+   /**
+    * Specifies the sku of the virtual machine scale set
+    */
+   public abstract VirtualMachineScaleSetSKU sku();
+
+   /**
+    * Specifies the tags of the virtual machine scale set
+    */
+   @Nullable
+   public abstract Map<String, String> tags();
+
+
+   /**
+    * Specifies the optional plan of the virtual machine scale set (only for market image)
+    */
+   @Nullable
+   public abstract VirtualMachineScaleSetPlan plan();
+
+   /**
+    * Specifies the properties of the availability set
+    */
+   @Nullable
+   public abstract VirtualMachineScaleSetProperties properties();
+
+   @SerializedNames({  "id", "name", "location", "sku", "tags", "plan", "properties"})
+   public static VirtualMachineScaleSet create(final String id, final String name, final String location,
+                                               VirtualMachineScaleSetSKU sku, final Map<String, String> tags,
+                                               VirtualMachineScaleSetPlan plan,
+                                               VirtualMachineScaleSetProperties properties) {
+      return builder().id(id).name(name).location(location).sku(sku).tags(tags)
+         .plan(plan).properties(properties)
+         .build();
+   }
+
+   public abstract Builder toBuilder();
+
+   private static Builder builder() {
+      return new AutoValue_VirtualMachineScaleSet.Builder();
+   }
+
+
+   @AutoValue.Builder
+   public abstract static class Builder {
+      public abstract Builder id(String id);
+      public abstract Builder name(String name);
+      public abstract Builder location(String location);
+      public abstract Builder sku(VirtualMachineScaleSetSKU sku);
+      public abstract Builder tags(Map<String, String> tags);
+      public abstract Builder plan(VirtualMachineScaleSetPlan plan);
+      public abstract Builder properties(VirtualMachineScaleSetProperties properties);
+
+      abstract Map<String, String> tags();
+      abstract VirtualMachineScaleSet autoBuild();
+
+      public VirtualMachineScaleSet build() {
+         tags(tags() == null ? ImmutableMap.<String, String>of() : ImmutableMap.copyOf(tags()));
```

Comment:
According to my previous comment, keep tags null if null, otherwise, make them immutable.